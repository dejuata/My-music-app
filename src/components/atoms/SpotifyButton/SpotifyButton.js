import React from 'react';
import PropTypes from 'prop-types';

export const SpotifyButton = ({ url }) => {
    return (
        <a
            href={ url }
            target="_blank"
            rel="noopener noreferrer"
        >
            <div className='btn btn-secondary'>
                Listen on Spotify
            </div>
        </a>
    )
}

SpotifyButton.propTypes = {
    url: PropTypes.string.isRequired
}
