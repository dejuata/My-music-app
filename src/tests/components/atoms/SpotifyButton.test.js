import React from 'react';
import { shallow } from 'enzyme';
import { SpotifyButton } from '../../../components/atoms/SpotifyButton/SpotifyButton';

describe('Test SpotifyButton', () => {

    const url = 'https://url-track-spotify';

    const wrapper = shallow(
        <SpotifyButton url={ url }/>
    )


    test('Should be render SpotifyButton', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Should be have the spotify track url', () => {
        const a = wrapper.find('a');

        expect( a.prop('href') ).toBe( url )
    })

})