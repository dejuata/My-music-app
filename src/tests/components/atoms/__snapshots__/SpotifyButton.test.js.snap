// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SpotifyButton Should be render SpotifyButton 1`] = `
<a
  href="https://url-track-spotify"
  rel="noopener noreferrer"
  target="_blank"
>
  <div
    className="btn btn-secondary"
  >
    Listen on Spotify
  </div>
</a>
`;
